src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -514,6 +518,10 @@ public CQLStatement prepare(ClientState state)
                 checkNotNull(select.parameters.refName, "Assignments must be named");
                 TxnDataName name = TxnDataName.user(select.parameters.refName);
                 checkTrue(selectNames.add(name), DUPLICATE_TUPLE_NAME_MESSAGE, name.name());
+                if (select.parameters.orderings != null && !select.parameters.orderings.isEmpty())
```

Comment:
Seems like maybe a bit of copy pasta forbidding things between `LET` selects and returning selects? Maybe the common forbidden things could be moved into a single shared method?